### DISCUSSION TITLE
Example needed of PDF that will not work with incremental save

### DISCUSSION THREAD

**Discussion by cmyers009** [pymupdf/PyMuPDF#3640] [Q&A] [answered]
As per the docs,
incremental (bool) – Only save changes to the PDF. Excludes “garbage” and “linear”. Can only be used if outfile is a string or a pathlib.Path and equal to Document.name. Cannot be used for files that are decrypted or repaired and also in some other cases. To be sure, check Document.can_save_incrementally(). If this is false, saving to a new file is required.

I do not have any examples of files that can not save incrementally.

Does anyone have an example PDF that I can use to test my error-handling logic?

**Comment by JorjMcKie** [comment]
That is easy: just make a 1-page empty file, edit it with some text editor and remove the last2 or 3 lines, then save.
When opening, MuPDF detects the missing end-of-file tag and starts a repair.
Repaired files cannot be saved incrementally.

**Comment by JorjMcKie** [comment, marked as answer]
Here is how:
```python
import pymupdf
doc=pymupdf.open()
page=doc.new_page()
doc.save("kaputt.pdf")
```
Gives you this:
```text
%PDF-1.7
%µ¶

1 0 obj
<</Type/Catalog/Pages 2 0 R>>
endobj

2 0 obj
<</Type/Pages/Count 1/Kids[4 0 R]>>
endobj

3 0 obj
<<>>
endobj

4 0 obj
<</Type/Page/MediaBox[0 0 595 842]/Rotate 0/Resources 3 0 R/Parent 2 0 R>>
endobj

xref
0 5
0000000000 00001 f 
0000000016 00000 n 
0000000062 00000 n 
0000000114 00000 n 
0000000135 00000 n 

trailer
<</Size 5/Root 1 0 R/ID[<2D1361C29C7F75306959677FC2A6C29A><0294DFE0C1FCB0F899F051FFB2CCF2F9>]>>
startxref
226
%%EOF
```
Delete the last 3 lines and save. Then open again and check whether we can save incrementally:
```python
doc=pymupdf.open("kaputt.pdf")
print(pymupdf.TOOLS.mupdf_warnings())
    format error: cannot find startxref
    trying to repair broken xref
    repairing PDF document
doc.is_repaired
True
doc.can_save_incrementally()
0
```
